src/middlewares/secretsManager.js

```diff
@@ -57,6 +58,9 @@ module.exports = opts => {
           // if we already have a cached secrets, then reset the timestamp so we don't
           // keep retrying on every invocation which can cause performance problems
           // when there's temporary problems with Secrets Manager
+          if (options.throwOnFailedCall) {
```

Comment:
I think we should only throw on the first request, if you have a cached value already you can continue with execution still. If we want to be smarter then we can build in some mechanism for the app developer to flush the cache - for example, by adding a function to the context object to `flushCachedSecrets`, but let's not do it in this PR.

Reply:
Hey @theburningmonk, 
I didn't thought of it, it's fixed now, lambda will fail only if first call didn't succeed